### PR TITLE
Square step2

### PR DESCRIPTION
**Pull request by kingbrendann**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ var Event = require('./models/event').Event;
 // CONTROLLERS
 // =============================================================================
 var feedItemController = require ('./controllers/feedItemController').FeedItemController;
+var listingController = require ('./controllers/listingController').EventController;
```
Comment (dodgertodd) [inline server.js]: .ListingController

**Comment by kingbrendann** [comment]
Ok. This should be good to go. Polling for items every hour. I need to properly page those results but I'm going to make that a [new issue](https://github.com/More2Life/m2l-server/issues/32) and do it later. 

Once we get the items response back from Square, we loop through the results and asynchronously check and update the listing. 

```
Logic:

if (matchingItemExistsInMongo) {
    if (lastUpdateDate of item in Mongo < lastUpdatedDate of item from Square) {
        update the document in Mongo
} else {
    create new Listing and save it to Mongo
}
```

Next step is to only request items from Square that have been updated since last poll. I'll make that a [separate issue](https://github.com/More2Life/m2l-server/issues/33) as well.

**Comment by kingbrendann** [comment]
Oh and I need to [update the RAML](https://github.com/More2Life/m2l-server/issues/34)

**Comment by kingbrendann** [comment]
### Complete
Polling only items updated since last poll:  https://github.com/More2Life/m2l-server/issues/33

**Comment by kingbrendann** [comment]
### Complete
Paginate items list response: https://github.com/More2Life/m2l-server/issues/32

File: controllers/listingController.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            redisController.setKeyValue(LAST_POLLED_AT_KEY, searchTime)
+        });
+    },
+
+    checkAndUpdateListing: async (item) => {
+        try {
+            console.log('ITEM TO SEARCH:');
+            console.log(item);
+
+            var listing = await Listing.findOne({'squareId' : item.id});
```
Comment (dodgertodd) [inline controllers/listingController.js:66]: JavaScript is madness.

**Review by dodgertodd — APPROVED** [review]

File: controllers/listingController.js (hunk 2 above)
Comment (kingbrendann) [reply controllers/listingController.js:66]: Don't you love it?!

**Pull request merged**